app/models/manager_refresh/inventory_collection.rb

```diff
@@ -441,7 +441,7 @@ def find(manager_uuid)
       when :local_db_find_missing_references
         data_index[manager_uuid] || find_in_db(manager_uuid)
       else
-        data_index[manager_uuid]
+        manager_uuid.kind_of?(Hash) ? find_by(manager_uuid) : data_index[manager_uuid]
```

Comment:
hm, and then it just works?

I was thinking about adding a lazy_find_by, so maybe we should rather remove the find_by methods and just check for the Hash in find methods?